### PR TITLE
Support caretPositionFromPoint for FF

### LINKED ISSUE #874: Unable to Edit Text in Windows on Firefox
## Bug Description

Unable to edit text in Win10/FFox. Clicked the "+" to add. Then tried to click within the text box to edit and am unable to.

Reported by Claudio in #576 

## Screenshots

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
Fixes #874

### CODE REVIEW

**Comment by spacedmonkey** [comment]
Can you provide any detail on this PR. Maybe a comment in the code to explain why this change is needed and what is fixes.

**Comment by miina** [comment]
@spacedmonkey Some browsers are using `caretPositionFromPoint`, such as FF, it was already existing in the code but the part of actually using it to return the value was missing. This PR just adds returning the value.

File: assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js [hunk 1]
```diff
@@ -44,6 +44,9 @@ function getCaretCharacterOffsetWithin(element, clientX, clientY) {
     if (clientX && clientY) {
       if (doc.caretPositionFromPoint) {
         range = document.caretPositionFromPoint(clientX, clientY);
+        if (range && range.offset) {
```
Comment (obetomuniz) [inline assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js]: nit: What about `if(range?.offset) { ... }`?

File: assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js [hunk 2]
```diff
@@ -44,6 +44,9 @@ function getCaretCharacterOffsetWithin(element, clientX, clientY) {
...
+          return range.offset;
```
Comment (barklund) [inline assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js]: This unfortunately doesn't work. There's a reason this function is as complicated, as it is. We need the character offset from the start of the full text container, not just the start of the node that the clicked text is inside.

Try making part of the text bold/italic/underline, defocus and the click inside the bold/italic/underline text. You'll see the offset registered as the offset of the nearest-node only: clicking between the 2nd and 3rd letter of a bold word will register as "offset=2" for the entire container, and then open the text editor with focus between the 2nd and 3rd letter of the entire text field overall. Which is very likely wrong.

Demo of current branch where editor consistently opens at the wrong offset in Firefox:
click-offset

What the original code did was: create a new selection all the way from the start of the container up to and including the same offset in the same node as discovered from the click, and the get the length of that selection in characters (across all selected nodes).

If this doesn't work in Firefox, we need to figure out a way to do the same thing in Firefox at least.

**Review by barklund — CHANGES_REQUESTED** [review]
It doesn't work for formatted text, unfortunately.

File: assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js (hunk 2 above)
Comment (miina) [reply assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js]: Eh, thanks for the information about the formatted text, didn't test with that!
Will look into it again.

Comment (miina) [reply assets/src/edit-story/utils/getCaretCharacterOffsetWithin.js]: @barklund The logic should be improved now, PTAL.

**Review by barklund — APPROVED** [review]
Works like a charm now!

**Review by spacedmonkey — APPROVED** [review]

**Pull request merged**